modules/loggers/mod_log_config.c

```diff
@@ -1229,16 +1242,42 @@ static int config_log_transaction(request_rec *r, config_log_state *cls,
     }
 
     for (i = 0; i < format->nelts; ++i) {
-        strs[i] = process_item(r, orig, &items[i]);
-        len += strl[i] = strlen(strs[i]);
+        lfd->portions[i] = process_item(r, orig, &items[i]);
+
+        /* if we don't have a custom log formatter,
+         * handle NULL, and empty string also escape all values unconditionally,
+         * else let the custom log formatter handle the escaping/formatting
+         */
+        if (!cls->log_formatter) {
+            if (!lfd->portions[i]) {
+                lfd->portions[i] = "-";
+            }
+            else if (!*lfd->portions[i]) {
+                lfd->portions[i] = "\"\"";
+            }
+            else {
+                if(items[i].tag) { /* or check for constant_item function, don't escape those */
+                    lfd->portions[i] = ap_escape_logitem(r->pool, lfd->portions[i]);
```

Comment:
possibly escaped twice for out-of-tree modules?  Do we have to remember if the registration happened via ap_register_log_handler so we don't escape again?

Reply:
good point, I'll add another parameter to the new ap_register_log_handler_ex to indicate if the handler does escape the return value.

Reply:
Hi @covener I added something regarding the double escaping, can you please have a look at https://github.com/apache/httpd/pull/429/commits/237c60b677e08aca501f275c7892abea170ddf1c

Reply:
MHH, one of the test workflow do fail not sure why. I'll check tomorrow:
https://github.com/thomasmey/httpd/actions/runs/9372712021/

Reply:
Hi, I found the bug and did fix the commit and did force push, all tests are green again, please review